target_snowflake/upload_clients/base_upload_client.py

```diff
@@ -24,3 +24,9 @@ def delete_object(self, stream: str, key: str) -> None:
         """
         Delete object
         """
+
+    @abstractmethod
+    def copy_object(self, source_key: str, target_key: str, target_metadata: dict) -> None:
```

Comment:
The [SnowflakeUploadClient](https://github.com/transferwise/pipelinewise-target-snowflake/blob/master/target_snowflake/upload_clients/snowflake_upload_client.py) doesn't implement this.

The `copy_object` acceptable only for s3_upload_client but I think an implementation is still required in the SnowflakeUploadClient as well. Maybe we can raise `NotImplementedError` with some human readable error message.